pkg/kubelet/winstats/winstats.go

```diff
@@ -0,0 +1,203 @@
+// +build windows
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package winstats provides a client to get node and pod level stats on windows
+package winstats
+
+import (
+	"context"
+	dockerapi "github.com/docker/engine-api/client"
+	cadvisorapi "github.com/google/cadvisor/info/v1"
+	cadvisorapiv2 "github.com/google/cadvisor/info/v2"
+	"os"
+	"runtime"
+	"sync"
+	"time"
+)
+
+// Client is an object that is used to get stats information.
+type Client struct {
+	dockerClient                *dockerapi.Client
+	cpuUsageCoreNanoSeconds     uint64
+	memoryPrivWorkingSetBytes   uint64
+	memoryCommittedBytes        uint64
+	mu                          sync.Mutex
+	memoryPhysicalCapacityBytes uint64
+}
+
+// NewClient constructs a Client.
+func NewClient() (*Client, error) {
+	client := new(Client)
+
+	dockerClient, _ := dockerapi.NewEnvClient()
+	client.dockerClient = dockerClient
+
+	// create physical memory
+	memory, err := getPhysicallyInstalledSystemMemoryBytes()
+
+	if err != nil {
+		return nil, err
+	}
+
+	client.memoryPhysicalCapacityBytes = memory
+
+	// start node monitoring (reading perf counters)
+	errChan := make(chan error, 1)
+	go client.startNodeMonitoring(errChan)
+
+	err = <-errChan
```

Comment:
Is there any benefit to do that? I don't use `startNodeMonitoring` or those channels anywhere else.